src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -68,10 +72,17 @@ private static double getActivePowerMismatch(Collection<LfBus> buses) {
 
     public DcLoadFlowResult run(Reporter reporter) {
         // only process main (largest) connected component
-        LfNetwork network = networks.get(0);
+        LfNetwork network = networks.stream().filter(n -> n.getNum() == ComponentConstants.MAIN_NUM).findAny().orElseThrow();
```

Comment:
I need the main synchronous component here.

Reply:
Can you create a method for that?

Reply:
After our discussion :  on lfNetwork I will use the existing connected component num and store the synchronous component num at loading, I will then update all filter I added to filter on CC num and SC num with MAIN_NUM value instead of just filter of CC num.